### PR TITLE
Feat get all categories

### PR DESCRIPTION
**Pull request by thegodhope**
you can check this one that was used for the previous cohort

I changed all the routes naming on product routes back to how it was according to the team lead’s  want. 
And also made necessary editing to the product docs to fit the naming.

- [x] My code follows the code style of this project.
- [x] This PR does not contain plagiarized content.
- [x] The title and description of the PR is clear and explains the approach.
- [x] I am making a pull request against the **main branch** (left side).
- [x] My commit messages styles matches our requested structure.
- [x] My code additions will fail neither code linting checks nor unit test.
- [x] I am only making changes to files I was requested to.

# Screenshots/ Videos

### CODE REVIEW

File: src/doc/swagger.json [hunk 1]
```diff
@@ -173,7 +173,7 @@
         }
       }
     },
-    "/products": {
```
Comment (Benrobo) [inline src/doc/swagger.json]: should be `products` and not `product`

**Review by Benrobo — CHANGES_REQUESTED** [review]

**Comment by thegodhope** [comment]
I have made the changes to the routes

File: src/doc/swagger.json [hunk 2]
```diff
@@ -219,7 +219,7 @@
         ]
       }
     },
-    "/products/nopromo": {
+    "/product/nopromo": {
```
Comment (Benrobo) [inline src/doc/swagger.json]: Pls leave this section this way with an `s`, I don't wanna allow the frontend to change the url and wait long time for build to get executed.

Comment (thegodhope) [reply src/doc/swagger.json]: done

**Pull request merged**